### DISCUSSION TITLE
Postgresql enum type

### DISCUSSION THREAD

**Discussion by waveskimmer** [micronaut-projects/micronaut-data#2390] [Q&A]
We'd like to create an enum type such as: 

```
create type primary_colors as ENUM ('Red', 'Yellow', 'Blue');

create table widget { 
    id                  UUID                                PRIMARY KEY,
   color             primary_colors NOT NULL
} 
```

A Text field works fine,  but I get this error with the above: 

_io.micronaut.data.exceptions.DataAccessException: Error executing PERSIST: ERROR: column "color" is of type enum_colors but expression is of type character varying
  Hint: You will need to rewrite or cast the expression.
  Position: 62_

But, cast to what?? 

Here's my Entity: 

```

enum class Color { RED, YELLOW, BLUE}

@MappedEntity(value = "widget", schema = "widgets")
data class Widget(
    @field:Id
    val widgetId: UUID = UUID.randomUUID(),
    @field:MappedProperty(converter = ColorConverter::class)
    val color: Color
)

@Singleton
class ColorConverter : AttributeConverter<Color, String> {
    override fun convertToPersistedValue(entityValue: Color?, context: ConversionContext): String? {
        return entityValue?.name!!
    }

    override fun convertToEntityValue(persistedValue: String?, context: ConversionContext): Color? {
        return Color.valueOf(persistedValue!!)
    }
}
```

I have this distilled into a small project.   Let me know if where to put it.

**Comment by dstepanov** [comment]
Please see an example https://github.com/micronaut-projects/micronaut-data/blob/master/data-jdbc/src/test/groovy/io/micronaut/data/jdbc/postgres/PostgresEnumSpec.groovy

**Reply by waveskimmer** [reply]
That would simplify my code, so thank you.  However, my problem in the database schema: 

```
create type primary_colors as ENUM ('Red', 'Yellow', 'Blue');

create table widget { 
    id                  UUID                                PRIMARY KEY,
   color             primary_colors NOT NULL
} 
```

**Reply by dstepanov** [reply]
Look correct like https://github.com/micronaut-projects/micronaut-data/blob/master/data-jdbc/src/test/groovy/io/micronaut/data/jdbc/postgres/PostgresDbInit.java

You might be missing `@MappedProperty(definition = "happiness", type = DataType.OBJECT)`

**Reply by waveskimmer** [reply]
I have my small example running now.   Thank you.
